Blink example comments / cleanup

ewenmcneill opened the pull request:
Remove unused (commented out) code.  Add comments hinting at what each section of the example code does (not all of it is directly related to the blinking LEDs, eg the PLL section).

There's still an inconsistency between the LED colour / `SB_RGBA_DRV` channel order for the `RGBnPWM` parameters and the `RGBn` parameters (supposedly the output of that module...), which I've not changed because so far, [at least currently with `nextpnr` it appears the mapping of `SB_RGBA_DRV` channels to output pins is fixed](https://github.com/YosysHQ/nextpnr/issues/254) -- and thus I've not been able to determine for sure whether the `pcf` files need correcting to match the actual circuit boards (as opposed to the schematics).  Eventually it'd be nice to (a) make it clearer if those parameters mean anything in practice and (b) if so, fix up the `pcf` files and list the parameters in an order consistent with the external hardware colours.

Feel free to update any of the comments if my understanding of the intended function is incorrect.

Ewen

ewenmcneill committed:
blink: describe function of each module section

ewenmcneill committed:
blink: Remove unused (commented out) code

xobs commented:
I appreciate the additional comments!  My only concern is about the removal of the HFOSC and LFOSC.  These were intended to demonstrate how to use these two blocks.

For example, right now I'm doing validation on DVT boards, and the crystal was placed on incorrectly.  I can copy the HFOSC example to get basically the same functionality, albeit at slightly worse timing.

Maybe the correct thing to do would be to pull the oscillators into their own test.

ewenmcneill commented:
@xobs, Without any comments those examples looked like they were old code which was then removed.  Which was why I removed them.

If you like I can re-push this branch (and thus this pull request) with just the first commit, possibly with some additional comments that those commented out bits of code are examples of testing the oscillators.

But yes, it might be cleaner to put the oscillators into their own test example, as they seem even less closely related to "blink LED" :-)

Let me know what you'd prefer.

Ewen

ewenmcneill commented:
@xobs As discussed on IRC, and in https://github.com/YosysHQ/nextpnr/issues/254#issuecomment-475840646 and https://github.com/YosysHQ/nextpnr/issues/254#issuecomment-475843155, it turnns out the mapping of `RGBn` outputs from `SB_RGBA_DRV` to physical pins is fixed (as identified by `arachne-pnr`), so I've swapped the order of the EVT3 `pcf` file to match what appears to have happened with the PCB layout (swapped red/blue).

From testing, the EVT2 `pcf` file is also wrong, and it has identical 39/40/41 (RGB0/RGB1/RGB2) connections to the LED.  So I've also swapped the EVT2 `pcf`to match the same `led_A` pin ordering.

Ewen

ewenmcneill committed:
blink: EVT3: swap led_r/led_b to match PCB layout

ewenmcneill committed:
pcf: swap EVT2 led_N pins to match PCB layout

Experimentally the EVT2 and EVT3 have identical circuit traces from
39/40/41 (RGB0/RGB1/RGB2) of ICE40UP5K-SG48 to the tri-colour LED,
so the colour mappings of the RGBn are identical.
